pkg/auth/ldaputil/client.go

```diff
@@ -78,3 +84,24 @@ func (l *LDAPClientConfig) Connect() (*ldap.Conn, error) {
 		return nil, fmt.Errorf("unsupported scheme %q", l.Scheme)
 	}
 }
+
+// Connect returns an established LDAP connection to which a bind has occurred if a bind DN and
+// password were given. If no error is returned, the caller is responsible for closing the connection
+func (l *LDAPClientConfig) Connect() (*ldap.Conn, error) {
```

Comment:
I disagree. If I am specifying a `bindDN` and a `bindPassword` it is assumed that I will connect *and* bind to the server.

I do *not* want to expose a `Bind()` method for the connection that does not bind if the credentials are not present, nor do I want to expose the credentials and expect the users to introspect and decide whether or not they need to bind. It makes the most  sense to keep this to the client configuration.